modules/programs/vscode.nix

```diff
@@ -0,0 +1,31 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.vscode;
+
+in
+
+{
+  options = {
+    programs.vscode = {
+      settings = mkOption {
```

Comment:
should we call this `userSettings` in case we want to add other kind of settings later?

Reply:
I don't think there exist any other settings except the per work space settings (which I don't think it makes remotely any sense to expose via the module). Any setting of vscode is contained in this file. If there are other options we provide, I would think they don't need to live in a settings attribute?

Reply:
That's true, I'm not a VScode user myself, I just felt since it's a user setting the name is more consistent and not ambiguous.